### PR TITLE
build: add CI for system tests

### PR DESCRIPTION
**Pull request by parthea**
This PR adds CI for system tests. Most of the configuration was taken from the split repository templates [here](https://github.com/googleapis/synthtool/tree/master/synthtool/gcp/templates/python_library/.kokoro).

### CODE REVIEW

File: .kokoro/system-single.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# `-e` enables the script to automatically fail when a command fails
+# `-o pipefail` sets the exit code to non-zero if any command fails,
+# or zero if all commands in the pipeline exit successfully.
+set -eo pipefail
+
+pwd
+
+# If NOX_SESSION is set, it only runs the specified session,
+# otherwise run all the sessions.
+if [[ -n "${NOX_SESSION:-}" ]]; then
```
Comment (vchudnov-g) [inline .kokoro/system-single.sh:25]: nit: I don't think the `:-` gets you anything here (cf https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion)


Also, what you have is fine, but you could equally well do 

```
SESSION_ARG=""
[[ -z "${NOX_SESSION} ]] || SESSION_ARG="-s ${NOX_SESSION}"
python3 -m nox ${SESSION_ARG}
```

This works in bash; in zsh, there's an additional tweak that's needed. I assume this file is executed, not sourced, in which case bash will always be used.

File: packages/google-cloud-access-approval/tests/system/smoke_test.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+@pytest.mark.parametrize("transport", ["grpc", "rest"])
+def test_list_approval_requests(project_id: str, transport: str):
+    client = accessapproval_v1.AccessApprovalClient(transport=transport)
+
+    parent = client.common_project_path(project_id)
+    approval_requests = list(client.list_approval_requests(parent=parent))
+
+    assert len(approval_requests) >= 0
```
Comment (vchudnov-g) [inline packages/google-cloud-access-approval/tests/system/smoke_test.py:36]: Just to confirm: the idea is that if the smoke test fails, we won't reach this line (i.e. we won't silently get a zero-length list)?

File: .kokoro/system.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+cd "$PROJECT_ROOT"
+
+pwd
+
+# A file for running system tests
+system_test_script="${PROJECT_ROOT}/.kokoro/system-single.sh"
+
+subdirs=(
+    ${PROJECT_ROOT}/packages
```
Comment (vchudnov-g) [inline .kokoro/system.sh:52]: I think you intended a trailing `*` here?

```suggestion
    ${PROJECT_ROOT}/packages/*
```

File: .kokoro/system.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+export PROJECT_ROOT=$(realpath $(dirname "${BASH_SOURCE[0]}")/..)
+
+cd "$PROJECT_ROOT"
+
+pwd
+
+# A file for running system tests
+system_test_script="${PROJECT_ROOT}/.kokoro/system-single.sh"
+
+subdirs=(
```
Comment (vchudnov-g) [inline .kokoro/system.sh:51]: nit:suggest s/`subdirs`/`package_dirs`/

File: .kokoro/system.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    ${PROJECT_ROOT}/packages
+)
+for subdir in ${subdirs[@]}; do
```
Comment (vchudnov-g) [inline .kokoro/system.sh:54]: Why do you have this outer loop? You don't seem to be using `subdir` anywhere. Your `find` command seems to be driving the loop.

File: .kokoro/system.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  # Run system tests for each package with directory packages/*/tests/system
+  for d in `find 'packages' -type d -wholename 'packages/*/tests/system'`; do
```
Comment (vchudnov-g) [inline .kokoro/system.sh:56]: suggest s/`d`/`dir`/

**Review by vchudnov-g — COMMENTED** [review]
Thanks for doing this!

One important comment about a wildcard, one non-trivial comment about the loop, and some minor comments.

File: .kokoro/system.sh (hunk 6 above)
Comment (parthea) [reply .kokoro/system.sh]: Fixed in 82f047f

File: .kokoro/system.sh (hunk 5 above)
Comment (parthea) [reply .kokoro/system.sh]: Fixed in 5775357

File: .kokoro/system.sh (hunk 4 above)
Comment (parthea) [reply .kokoro/system.sh]: Obsolete. Code was deleted in 5775357

File: .kokoro/system.sh (hunk 3 above)
Comment (parthea) [reply .kokoro/system.sh]: Obsolete. Code was deleted in https://github.com/googleapis/google-cloud-python/commit/5775357cadfa73c500c7df52de66bcc09296088d

File: packages/google-cloud-access-approval/tests/system/smoke_test.py (hunk 2 above)
Comment (parthea) [reply packages/google-cloud-access-approval/tests/system/smoke_test.py]: Clarified the purpose of the smoke test in [4ebbaf5](https://github.com/googleapis/google-cloud-python/pull/11735/commits/4ebbaf58da40fa7c58da3b3dde83ccf8ef5cb4dd)

File: .kokoro/system-single.sh (hunk 1 above)
Comment (parthea) [reply .kokoro/system-single.sh]: Fixed in 68e60f2

File: packages/google-cloud-access-approval/tests/system/smoke_test.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+
+@pytest.mark.parametrize("transport", ["grpc", "rest"])
+def test_list_approval_requests(project_id: str, transport: str):
+    client = accessapproval_v1.AccessApprovalClient(transport=transport)
+
+    parent = client.common_project_path(project_id)
+    client.list_approval_requests(parent=parent)
+
+    # The purpose of this smoke test is to test the communication with the API,
```
Comment (vchudnov-g) [inline packages/google-cloud-access-approval/tests/system/smoke_test.py:36]:
```suggestion
    # The purpose of this smoke test is to test the communication with the API server,
```

**Review by vchudnov-g — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**